src/vma/dev/ring.h

```diff
@@ -121,47 +121,43 @@ class ring
 	virtual bool		is_ratelimit_supported(struct vma_rate_limit_t &rate_limit) = 0;
 	virtual int		reg_mr(void *addr, size_t length, uint32_t &lkey) { NOT_IN_USE(addr); NOT_IN_USE(length); NOT_IN_USE(lkey); return -1;};
 	virtual int		dereg_mr(void *addr, size_t length) { NOT_IN_USE(addr);NOT_IN_USE(length); return -1;};
-#ifdef DEFINED_SOCKETXTREME		
+
```

Comment:
I see no reason why socketxtreme_poll() should be declared here. why not moving it to ring_simple?
cq_channel_info::get_ring() should return ring_simple object.